File: Cpp/991_Broken_Calculator.cpp
class Solution {
public:
    int brokenCalc(int startValue, int target) {
        int result = 0;
        int current = target;
        
        while(current > startValue){
            if(current %2 == 1){
                ++result;
                ++current;
            }
            ++result;
            current /= 2;
        }
        result += (startValue - current);
        
        return result;
    }
};

File: Kotlin/991_Broken_Calculator.kt
class Solution {
    fun brokenCalc(startValue: Int, target: Int): Int {
        var result = 0
        var current = target
        
        while(current > startValue){
            if(current % 2 == 1){
                ++result
                ++current
            }
            ++result
            current /= 2
        }
        result += (startValue - current)
        
        return result
    }
}